main/src/main/scala/sbt/Keys.scala

```diff
@@ -162,8 +162,8 @@ object Keys {
 
   // Output paths
   val classDirectory = SettingKey[File]("class-directory", "Directory for compiled classes and copied resources.", AMinusSetting)
-  val cleanFiles = SettingKey[Seq[File]]("clean-files", "The files to recursively delete during a clean.", BSetting)
-  val cleanKeepFiles = SettingKey[Seq[File]]("clean-keep-files", "Files to keep during a clean.", CSetting)
+  val cleanFiles = TaskKey[Seq[File]]("clean-files", "The files to recursively delete during a clean.", BSetting)
```

Comment:
We now have a criteria for setting vs task: make it a task makes it un-queryable in server mode. Ideally this would be a `SettingKey[PathFinder]` that we evaluate during clean. Should we change to that?